spec/system/articles/user_visits_articles_by_timeframe_spec.rb

```diff
@@ -1,9 +1,10 @@
 require "rails_helper"
 
-RSpec.describe "User visits articles by timeframe" do
+# rubocop:disable RSpec/ExampleLength
+RSpec.describe "User visits articles by timeframe", js: true do
```

Comment:
Merge these individual tests into two large tests. It makes sense to do this with system test as it mimics what user might do and speed up the test.